metadata_validation_conversion/ontology_improver/tasks.py

```diff
@@ -0,0 +1,96 @@
+import json
+import requests
+import pandas as pd
+from collections import Counter
+from django.conf import settings
+from elasticsearch import Elasticsearch, RequestsHttpConnection
+from metadata_validation_conversion.celery import app
+
+es = Elasticsearch([settings.NODE], connection_class=RequestsHttpConnection, \
+                    http_auth=(settings.ES_USER, settings.ES_PASSWORD), \
+                    use_ssl=True, verify_certs=False)
+
+def type_count(data):
+    data = data.split(', ')
+    data = dict(Counter(data))
+    return data
+
+def convertToListOfDict(data):
+    l = []
+    for k in data:
+        obj = {
+            'type': k,
+            'count': data[k]
+        }
+        l.append(obj)
+    return l
+
+def comma_separated_combine(data):
+    values = []
+    for i in data:
+        i_list = i.split(', ')
+        values = values + i_list
+    return ', '.join(values)
+
+def get_species_for_project(project):
+    project_filter = json.dumps({
+        'secondaryProject': [project]
+    })
+    r = requests.get(f'http://backend-svc:8000/data/organism/_search/?size=10000&filters={project_filter}')
+    data = json.loads(r.content)['hits']['hits']
+    species = filter(lambda y: y, set(map(lambda x : x['_source']['organism']['text'], data)))
+    species = ', '.join(species)
+    return species
+
+@app.task
+def update_ontology_summary():
+    r = requests.get(f'http://backend-svc:8000/data/ontologies/_search/?size=10000')
+    if r.status_code == 200:
+        ontologies = json.loads(r.content)['hits']['hits']
+        ontologies = map(lambda ontology : ontology['_source'], ontologies)
+        df = pd.DataFrame.from_dict(ontologies)[['projects', 'type', 'term']]
+        df['type'] = [', '.join(map(str, l)) for l in df['type']]
+        df = df.explode('projects')
+        valid_projects = [
+            "AQUA-FAANG",
+            "BovReg",
+            "GENE-SWitCH",
+            "Bovine-FAANG",
+            "EFFICACE",
+            "GEroNIMO",
+            "RUMIGEN",
+            "Equine-FAANG"
+        ]
+        df = df.loc[df['projects'].isin(valid_projects)]
+        df = df.groupby(['projects']).agg({
+            'type': comma_separated_combine,
+            'term': 'count'
+        }).reset_index()
+        df['type'] = df['type'].apply(type_count)
+        # get project-specific species from organisms index
+        species = {}
+        for project in valid_projects:
+            if project in df['projects'].values:
+                species[get_species_for_project(project)] = project
+        df['species'] = species
+        # get existing summary statistics
+        url = f"http://backend-svc:8000/data/summary_ontologies/_search/?size=10"
```

Comment:
would be great to read URL value from global constants

Reply:
moved to BE_SVC constant in metadata_validation_conversion/metadata_validation_conversion/constants.py